antlr-kotlin-runtime/src/commonMain/kotlin/org/antlr/v4/kotlinruntime/dfa/DFA.kt

```diff
@@ -89,12 +68,12 @@ public open class DFA(
    * @return The start state corresponding to the specified precedence,
    *   or `null` if no start state exists for the specified precedence
    *
-   * @throws IllegalStateException if this is not a precedence DFA
+   * @throws IllegalStateException If this is not a precedence DFA
    * @see isPrecedenceDfa
    */
   public fun getPrecedenceStartState(precedence: Int): DFAState? {
-    check(isPrecedenceDfa) {
-      "Only precedence DFAs may contain a precedence start state."
+    if (!isPrecedenceDfa) {
```

Comment:
Could we use `require`?

Reply:
The reason I'm using an explicit `IllegalStateException` is matching the Java runtime, and overall uniformity.

Or we choose to use `check`/`require` everywhere, or we stick to explicit exceptions.

Reply:
Understood